client/landing/stepper/declarative-flow/internals/steps-repository/design-setup/index.tsx

```diff
@@ -108,6 +108,14 @@ const designSetup: Step = function DesignSetup( { navigation, flow } ) {
 
 	const visibility = useNewSiteVisibility();
 
+	const hasTrackedView = useRef( false );
+	useEffect( () => {
+		if ( showGeneratedDesigns && ! hasTrackedView.current ) {
+			hasTrackedView.current = true;
+			recordTracksEvent( 'calypso_signup_generated_design_picker_view' );
```

Comment:
Suggestion: Would be nice if we added
(1) vertical_id, and
(2) the order of the generated designs (comma separated string of the recipe slugs) so that we know what did the customer see first.

Reply:
Maybe we can add `onMount` prop to the Generated Design Picker and then we can just call `recordTracksEvent` in that function without checking anything.

Reply:
I think if I did an `onMount` prop, I would have to implement that myself since it isn't implemented by FunctionalComponent. It is in effect implemented as a `useEffect` block https://stackoverflow.com/questions/71434321/react-functional-component-how-to-use-componentdidmount. But then if I want to ensure that the event is not re-sent when using the back button from the static previews, I would probably still need to add a similar useRef trick and wind up with the original code 🤷.

Reply:
Oh yes, you're right. I forgot we don't want to track when people go back to generated design picker 😔

Reply:
@taipeicoder I updated it to send vertical_id and a generated_designs which is a list of slugs sepperated by comas  as you suggested 👍